games-arcade/edgar/patches/edgar-1.30.patchset

```diff
@@ -1,40 +1,81 @@
-From 04f238abd162bc62a3d4bd846ef34d644c5e9f3b Mon Sep 17 00:00:00 2001
-From: begasus <begasus@gmail.com>
-Date: Fri, 4 Jan 2019 08:58:59 +0100
-Subject: correct load/save path
+From 4c613e2d714402349620ebf6543952ef403ae566 Mon Sep 17 00:00:00 2001
+From: TURX <turx2003@gmail.com>
+Date: Tue, 21 Jan 2020 23:14:22 +0800
+Subject: [PATCH] fix Haiku support
 
+---
+ src/system/load_save.c | 24 ++++++++++++++++++++++++
+ 1 file changed, 24 insertions(+)
 
 diff --git a/src/system/load_save.c b/src/system/load_save.c
-index 2d55feb..b6b136d 100644
+index 2d55feba5..8036cad12 100644
 --- a/src/system/load_save.c
 +++ b/src/system/load_save.c
-@@ -73,7 +73,7 @@ extern Game game;
+@@ -41,6 +41,11 @@ Foundation, 51 Franklin Street, Suite 500, Boston, MA 02110-1335, USA.
+ #include "load_save.h"
+ #include "resources.h"
+ 
++#ifdef __HAIKU__
++#include <FindDirectory.h>
++#include <string.h>
++#endif
++
+ static char gameSavePath[MAX_PATH_LENGTH], tempFile[MAX_PATH_LENGTH], saveFileIndex[MAX_PATH_LENGTH], continueFile[MAX_PATH_LENGTH];
+ static int temporaryDataExists;
+ 
+@@ -57,6 +62,7 @@ extern Game game;
+ 	{
+ 		char *userHome;
+ 		char dir[MAX_PATH_LENGTH];
++		char path[MAXPATHLEN];
+ 		struct passwd *pass;
+ 
+ 		pass = getpwuid(geteuid());
+@@ -73,8 +79,14 @@ extern Game game;
  		#if MACOS == 1
  		snprintf(dir, sizeof(dir), "%s/Library/Application Support", userHome);
  		#else
--		snprintf(dir, sizeof(dir), "%s/.parallelrealities", userHome);
-+		snprintf(dir, sizeof(dir), "%s/config/settings/parallelrealities", userHome);
++		#ifdef __HAIKU__
++		path[MAXPATHLEN] = 0;
```

Comment:
```suggestion
```

There is no need for this, `find_directory` will terminate the path with a `\0`, and it won't exceed the size you provided. Also the code is wrong since you're overstepping the bounds by a byte (max index of array is length - 1)